src/librustc_trans/collector.rs

```diff
@@ -850,8 +910,14 @@ impl<'b, 'a, 'v> ItemLikeVisitor<'v> for RootCollector<'b, 'a, 'v> {
                 // const items only generate translation items if they are
                 // actually used somewhere. Just declaring them is insufficient.
             }
-            hir::ItemFn(.., ref generics, _) => {
-                if !generics.is_type_parameterized() {
+            hir::ItemFn(_, _, constness, _, ref generics, _) => {
+                let is_const = match constness {
```

Comment:
I recommend `tcx.is_const_fn(def_id)` - similarly for `hir::Generics`, `ty::Generics` should be preferred, although that one is preexisting.